AVX512 support

isuruf opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

isuruf committed:
AVX512 support

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ The recipe is not parsable by parser `conda-souschef (grayskull)`. This parser is not currently used by conda-forge, but may be in the future. We are collecting information to see which recipes are compatible with grayskull.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/16729133202](https://github.com/conda-forge/conda-forge-webservices/actions/runs/16729133202). Examine the logs at this URL for more detail.</sub>

isuruf committed:
build executables and externals with v3

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I was trying to look for recipes to lint for you, but it appears we have a merge conflict. Please try to merge or rebase with the base branch to resolve this conflict.

Please ping the 'conda-forge/core' team (using the `@` notation in a comment) if you believe this is a bug.

isuruf committed:
Merge branch 'main' of github.com:conda-forge/tensorflow-feedstock into avx512

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ The recipe is not parsable by parser `conda-souschef (grayskull)`. This parser is not currently used by conda-forge, but may be in the future. We are collecting information to see which recipes are compatible with grayskull.
 * ℹ️ The recipe is not parsable by parser `conda-recipe-manager`. The recipe can only be automatically migrated to the new v1 format if it is parseable by conda-recipe-manager.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/22456640361](https://github.com/conda-forge/conda-forge-webservices/actions/runs/22456640361). Examine the logs at this URL for more detail.</sub>

isuruf committed:
MNT: Re-rendered with conda-smithy 3.55.1 and conda-forge-pinning 2026.02.19.14.04.59

Other tools:
- conda-build 25.9.0
- rattler-build 0.47.1
- rattler-build-conda-compat 1.4.5

isuruf committed:
remove 12.8 migrator

isuruf committed:
commit removal

isuruf committed:
fix test

h-vetinari commented on recipe/meta.yaml:
```diff
@@ -395,7 +395,7 @@ outputs:
         - cat test_tensorflow.py     # [unix]
         - run python test_tensorflow.py  # [build_platform == target_platform]
-        - tf_upgrade_v2 --help       # [build_platform == target_platform]
+        - run tf_upgrade_v2 --help       # [build_platform == target_platform]
```
This didn't fix the test:
```
/home/conda/feedstock_root/build_artifacts/tensorflow-split_1771543504012/test_tmp/run_test.sh: line 10: run: command not found
```

h-vetinari replied:
While you're debugging, please restrict testing here to one (preferably non-CUDA) build.

h-vetinari commented on .ci_support/migrations/cuda128.yaml:
```diff
@@ -1,52 +0,0 @@
-    cuda_compiler_version:
-      - 11.8
-      - 12.4
-      - 12.6
-      - None
-      - 12.8
-      - 12.9
```
this migrator should not be deleted, we still need 12.8 here (until 2.20 I guess?); it already has `use_local: true` so you can simply move 12.9 before 12.8 in the ordering here. and a rerender should do the correct thing (== stay on 12.8)

h-vetinari commented:
As an overall point; not sure if we're doing AVX512 builds anywhere else yet? Are there some benchmarks that we can reference for how much of a win this is over AVX2? Makes a big difference if it's 5% or 50%.

isuruf committed:
fix emulation

isuruf committed:
remove tensorflow-estimator

isuruf committed:
skip level = 1

h-vetinari commented:
> While you're debugging, please restrict testing here to one (preferably non-CUDA) build.

This is not a difficult demand. I'm cancelling the builds.

isuruf commented:
I skipped level=1, but since level=3 and level=4 failed and  cuda vs non-cuda failed with different errors, I wanted to check all 4 builds.

h-vetinari commented:
We have a pending pytorch merge that fixes several issues, which is IMO pretty self-evidently of higher priority than debugging work, especially for features rather than fixes. The PR here would block the server for 8-12h as-is, which I think is not reasonable under the circumstances. Though you should be able to run them in parallel if you're willing to try the cirrus setup from https://github.com/conda-forge/pytorch-cpu-feedstock/pull/484

h-vetinari commented:
I had restarted all builds here as soon as runners became available again ~36h ago, but the interface with cirun is pretty fragile these days, and only one job started. The only recovery is to cancel and restart, but since the GH UI has no third way between "restart all failed jobs" and "restart one job", the former would have meant rerunning the just-failed job again. In any case, all the last of the 4 jobs whose output you wanted to see has now been started, 3 are done already.

isuruf commented:
Moved some fixes from this PR to https://github.com/conda-forge/tensorflow-feedstock/pull/477
